pkg/accounting/tally/tally_test.go

```diff
@@ -30,91 +31,162 @@ import (
 	pointerdb := pointerdb.NewServer(teststore.New(), &overlay.Cache{}, logger, pointerdb.Config{}, nil)
 
-	const N = 50
 	nodes := []*pb.Node{}
 	nodeIDs := storj.NodeIDList{}
 	expectedOnline := []*pb.Node{}
-	for i := 0; i < N; i++ {
-		nodeID := teststorj.NodeIDFromString(strconv.Itoa(i))
-		n := &pb.Node{Id: nodeID, Type: pb.NodeType_STORAGE, Address: &pb.NodeAddress{Address: ""}}
-		nodes = append(nodes, n)
+	var expectedNodeMap = make(map[string]int64)
+	ns := planet.StorageNodes
+
+	for i, n := range ns {
+		node := &pb.Node{Id: n.ID(), Type: pb.NodeType_STORAGE, Address: &pb.NodeAddress{Address: n.Addr()}}
+		nodes = append(nodes, node)
 		if i%(rand.Intn(5)+2) == 0 {
-			id := teststorj.NodeIDFromString("id" + nodeID.String())
+			id := teststorj.NodeIDFromString("id" + n.ID().String())
 			nodeIDs = append(nodeIDs, id)
+			expectedNodeMap[id.String()] = 0
 		} else {
-			nodeIDs = append(nodeIDs, nodeID)
-			expectedOnline = append(expectedOnline, n)
+			nodeIDs = append(nodeIDs, n.ID())
+			expectedOnline = append(expectedOnline, node)
 		}
 	}
+
 	overlayServer := mocks.NewOverlay(nodes)
-	kad := &kademlia.Kademlia{}
 	limit := 0
 	interval := time.Second
 
-	accountingDb, err := accounting.NewDb("sqlite3://file::memory:?mode=memory&cache=shared")
+	accountingDb, err := accounting.NewDB("sqlite3://file::memory:?mode=memory&cache=shared")
 	assert.NoError(t, err)
 	defer func() { _ = accountingDb.Close() }()
 
 	bwDb, err := dbManager.NewDBManager("sqlite3", "file::memory:?mode=memory&cache=shared")
 	assert.NoError(t, err)
 	defer func() { _ = accountingDb.Close() }()
-
-	tally := newTally(logger, accountingDb, bwDb, pointerdb, overlayServer, kad, limit, interval)
-
-	online, err := tally.onlineNodes(ctx, nodeIDs)
+	tally, err := newTally(ctx, logger, accountingDb, bwDb, pointerdb, overlayServer, kad, limit, interval)
+	assert.NoError(t, err)
+	online, nodeMap, err := tally.categorize(ctx, nodeIDs)
 	assert.NoError(t, err)
 	assert.Equal(t, expectedOnline, online)
+	assert.Equal(t, expectedNodeMap, nodeMap)
 }
 
 func TestTallyAtRestStorage(t *testing.T) {
+	ctx := testcontext.New(t)
+	defer ctx.Cleanup()
 
-}
+	planet, err := testplanet.New(t, 1, 30, 0)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer ctx.Check(planet.Shutdown)
 
-func TestNeedToContact(t *testing.T) {
+	planet.Start(ctx)
 
-}
+	kad := planet.Satellites[0].Kademlia
+	pointerdb := pointerdb.NewServer(teststore.New(), &overlay.Cache{}, zap.NewNop(), pointerdb.Config{}, nil)
+	overlayServer := mocks.NewOverlay([]*pb.Node{})
+	accountingDb, err := accounting.NewDB("sqlite3://file::memory:?mode=memory&cache=shared")
+	assert.NoError(t, err)
+	defer func() { _ = accountingDb.Close() }()
+	bwDb, err := dbManager.NewDBManager("sqlite3", "file::memory:?mode=memory&cache=shared")
+	assert.NoError(t, err)
+	defer func() { _ = accountingDb.Close() }()
+	tally, err := newTally(ctx, zap.NewNop(), accountingDb, bwDb, pointerdb, overlayServer, kad, 0, time.Second)
```

Comment:
tally really shouldn't need kademlia, is there something else missing?

Reply:
Tally needs access to the node client to ping nodes. We need kademlia to init this node client. Is there a better way?